packages/melos/test/commands/run_test.dart

```diff
@@ -631,6 +635,338 @@ melos run hello_script
+
+2 issues found.
+a: SUCCESS
+--------------------------------------------------------------------------------
+b:
+Analyzing b...
+No issues found!
+b: SUCCESS
+--------------------------------------------------------------------------------
+c:
+Analyzing c...
+No issues found!
+c: SUCCESS
+--------------------------------------------------------------------------------
+
+\$ melos analyze
+  └> dart analyze 
+     └> SUCCESS
+
+melos run hello_script
+  └> analyze
+     └> SUCCESS
+
+melos run hello_script
+  └> echo "hello world"
+     └> RUNNING
+
+${currentPlatform.isWindows ? '"hello world"' : 'hello world'}
+
+melos run hello_script
+  └> echo "hello world"
+     └> SUCCESS
+
+''',
+        ),
+      );
+    });
+
+    test(
+        'verifies that a Melos script can call another script containing '
+        'a script with a name equal to a melos command,  and ensures the '
+        'script group successfully runs instead of the command', () async {
+      final workspaceDir = await createTemporaryWorkspace(
+        runPubGet: true,
+        configBuilder: (path) => MelosWorkspaceConfig(
+          path: path,
+          name: 'test_package',
+          packages: [
+            createGlob('packages/**', currentDirectoryPath: path),
+          ],
+          scripts: const Scripts({
+            'hello_script': Script(
+              name: 'hello_script',
+              steps: ['analyze', 'echo "hello world"'],
+            ),
+            'analyze': Script(
+              name: 'hello_script',
+              run: 'dart analyze . --fatal-warnings',
+            ),
+          }),
+        ),
+      );
+
+      aDir = await createProject(
+        workspaceDir,
+        const PubSpec(name: 'a'),
+      );
+
+      await createProject(
+        workspaceDir,
+        const PubSpec(name: 'b'),
+      );
+
+      await createProject(
+        workspaceDir,
+        const PubSpec(
+          name: 'c',
+        ),
+      );
+
+      writeTextFile(
+        p.join(aDir.path, 'main.dart'),
+        r'''
+        void main() {
+          for (var i = 0; i < 10; i++) {
+            print('hello ${i + 1}');
+          }
+        }
+      ''',
+      );
+
+      final logger = TestLogger();
+      final config = await MelosWorkspaceConfig.fromWorkspaceRoot(workspaceDir);
+      final melos = Melos(
+        logger: logger,
+        config: config,
+      );
+
+      await melos.run(scriptName: 'hello_script', noSelect: true);
+
+      expect(
+        logger.output.normalizeNewLines(),
+        ignoringAnsii(
+          '''
+melos run hello_script
+  └> analyze
+     └> RUNNING
+
+melos run analyze
+  └> dart analyze . --fatal-warnings
+     └> RUNNING
+
+Analyzing ....
+
+   info - ${currentPlatform.isWindows ? 'packages\a\main.dart' : 'packages/a/main.dart'}:3:13 - Don't invoke 'print' in production code. Try using a logging framework. - avoid_print
```

Comment:
Aah the pain of Windows, this has to have two backslashes `\\`

Reply:
Ohhh I see, hopefully the last commit solves this issue!

Reply:
for some reason, even though I added the two backslashes, the error seems to be something else @spydon:

```console
              'warning - pubspec.yaml:6:11 - The path \'d:\\a\\melos\\melos\\packages\\melos\' isn\'t a POSIX-style path. Try converting the value to a POSIX-style path. - path_not_posix\n'
              '   info - packages\\a\\main.dart:3:13 - Don\'t invoke \'print\' in production code. Try using a logging framework. - avoid_print\n'
              '   info - packages\\a\\main.dart:5:10 - Missing a newline at the end of the file. Try adding a newline at the end of the file. - eol_at_end_of_file\n'
              '\n'
              '3 issues found.\n'

```

Reply:
@spydon I updated the test and use a `list` script instead of the `analyze` one to avoid all these issues related to windows environments, and still testing the right scenario, which is prioritazing custom scripts instead of Melos commands when scripts with same names as the commands are used. commit e1d2e9655dcf6dfc5ed7cab7217fa2a8906a8fd5